codecparsers/jpegParser.cpp

```diff
@@ -313,6 +313,9 @@ bool Parser::nextMarker()
     const uint8_t* current = start;
     const uint8_t* match;
 
```

Comment:
hi @uartie , if m_size = 0, the start will be m_data - 1. 
Could we adjust code just let
>const uint8_t* const end = m_data + m_size;

we only need change following lines
>if (match == end || (match +1) == end)
>      return false;